### ISSUE TITLE
Since 2.53: Unable to use custom Firefox path in (non-remote) WebDriver

### ISSUE THREAD

**Issue by icemac** [SeleniumHQ/selenium#1965]
## Meta -
OS: any
Selenium Version: since 2.53, Python bindings
Browser: Firefox
Browser Version: any

## Expected Behavior -

When calling `selenium.webdriver.firefox.webdriver.WebDriver()` with a custom path to a Firefox binary it should use this one instead the default one.

## Actual Behavior -

The setting is either silently ignored or produces a traceback.

## Steps to reproduce -

* Using `FirefoxBinary` is ignored:

  ```
from selenium.webdriver.firefox.firefox_binary import FirefoxBinary

selenium.webdriver.firefox.webdriver.WebDriver(firefox_binary= FirefoxBinary('/path/to/my/firefox'))
```
  Because of https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/firefox/webdriver.py#L67
and `_get_firefox_start_cmd()` ignoring the path set in `__init__`, see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/firefox/firefox_binary.py#L142

* Using a plain path `selenium.webdriver.firefox.webdriver.WebDriver(firefox_binary='/path/to/my/firefox')` fails:

   ```
  File ".../selenium/webdriver/firefox/webdriver.py", line 103, in __init__
    self.binary, timeout)
  File ".../selenium/webdriver/firefox/extension_connection.py", line 51, in __init__
    self.binary.launch_browser(self.profile, timeout=timeout)
AttributeError: 'str' object has no attribute 'launch_browser'
```

* Using `firefox_options` as argument: 

  ```
from selenium.webdriver.firefox.options import Options
from selenium.webdriver.firefox.firefox_binary import FirefoxBinary

options = Options()
options.binary_location = FirefoxBinary('/path/to/my/firefox')
selenium.webdriver.firefox.webdriver.WebDriver(firefox_options=options)
```
  fails:
  ```
 File ".../selenium/webdriver/firefox/webdriver.py", line 107, in __init__
    keep_alive=True)
  File ".../selenium/webdriver/remote/webdriver.py", line 91, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File ".../selenium/webdriver/remote/webdriver.py", line 173, in start_session
    'desiredCapabilities': desired_capabilities,
  File ".../selenium/webdriver/remote/webdriver.py", line 231, in execute
    response = self.command_executor.execute(driver_command, params)
  File ".../selenium/webdriver/remote/remote_connection.py", line 392, in execute
    data = utils.dump_json(params)
  File ".../selenium/webdriver/remote/utils.py", line 32, in dump_json
    return json.dumps(json_struct)
  File ".../lib/python2.7/json/__init__.py", line 244, in dumps
    return _default_encoder.encode(obj)
  File ".../lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File ".../lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File ".../lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <selenium.webdriver.firefox.firefox_binary.FirefoxBinary object at 0x106732350> is not JSON serializable
```

**Comment by gunesacar** [comment]
This bug seems to be fixed in Selenium 2.53.3.
Tests to confirm: https://gist.github.com/gunesacar/27864fd32d00da4532a5af45d43902a0

**Comment by lukeis** [comment]
this should be all fixed... please raise a new issue if you still have issues.
